staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -786,29 +792,51 @@ type WatchListResult struct {
 	// the end of the stream.
 	initialEventsEndBookmarkRV string
 
-	// gv represents the API version
-	// it is used to construct the final list response
-	// normally this information is filled by the server
-	gv schema.GroupVersion
+	// negotiatedObjectDecoder knows how to decode
+	// the initialEventsListBlueprint
+	negotiatedObjectDecoder runtime.Decoder
+
+	// base64EncodedInitialEventsListBlueprint contains an empty,
+	// versioned list encoded in the requested format
+	// (e.g., protobuf, JSON, CBOR) and stored as a base64-encoded string
+	base64EncodedInitialEventsListBlueprint string
 }
 
 func (r WatchListResult) Into(obj runtime.Object) error {
 	if r.err != nil {
 		return r.err
 	}
 
-	listPtr, err := meta.GetItemsPtr(obj)
+	listPointer, err := conversion.EnforcePtr(obj)
+	if err != nil {
+		return err
+	}
+	listItemsPtr, err := meta.GetItemsPtr(obj)
 	if err != nil {
 		return err
 	}
-	listVal, err := conversion.EnforcePtr(listPtr)
+	listVal, err := conversion.EnforcePtr(listItemsPtr)
 	if err != nil {
 		return err
 	}
 	if listVal.Kind() != reflect.Slice {
 		return fmt.Errorf("need a pointer to slice, got %v", listVal.Kind())
 	}
 
+	encodedInitialEventsListBlueprint, err := base64.StdEncoding.DecodeString(r.base64EncodedInitialEventsListBlueprint)
+	if err != nil {
+		return err
+	}
+	initialEventsListBlueprint, _, err := r.negotiatedObjectDecoder.Decode(encodedInitialEventsListBlueprint, nil, obj)
+	if err != nil {
+		return err
+	}
+	initialEventsListBlueprintValue := reflect.ValueOf(initialEventsListBlueprint).Elem()
```

Comment:
I think this is okay. We do have something similar for adding items to the final list.

Reply:
You may be able to avoid it entirely. When you pass a non-nil "into" argument to Decode, the object it returns isn't guaranteed to be the same as the argument in all cases, but I can't think of a realistic case where `obj` and `initialEventsListBlueprint` have the same concrete pointer type here but are not pointers to the same object.

I see that `(Result).Into` effectively swallows any case that would cause Decode to return a nil error and an object other than the provided "into" argument. The safest thing to do if that happens is to return an error. Otherwise, callers will assume that the object they passed to `Into` has been populated from the response body when that may not be true.

Reply:
I don't mind performing an additional check on the list types. I would like to be absolutely sure that the lists are of the same type.

Is there anything wrong with that?

Reply:
The last commit contains the changes you suggested. PTAL. Thanks.